pkg/router/route_group.go

```diff
@@ -164,21 +165,47 @@ func (rg *RouteGroup) Write(p []byte) (n int, err error) {
 	}
 
 	rg.mu.Lock()
-	tp, err := rg.tp()
-	if err != nil {
-		rg.mu.Unlock()
-		return 0, err
+	tps := make([]*transport.ManagedTransport, len(rg.tps))
```

Comment:
I know that we've been doing it this way for a long time. but what do you think if we get rid of mutex usage and copying in here? afaik we initialize `tps` and `fwd` at the point of route group creation. and after that these slices are actually not changed. also `rvs` is a subject for discussion, it seems like it's not used at all

Reply:
@Darkren Good idea